### PR TITLE
lower resultsdb reporting into test-override job

### PR DESCRIPTION
**Pull request by dustymabe**
This way one can easily replay ↺ a job to retry a failed test and still have the results reported back to Bodhi.

Should resolve https://github.com/coreos/coreos-ci/issues/64

### CODE REVIEW

File: jobs/test-override.Jenkinsfile [hunk 1]
```diff
@@ -241,4 +237,38 @@ try {
...
+                shwrap("""/usr/lib/coreos-assembler/resultsdb-report   \
+                    --testcase cosa.build-and-test                     \
+                    --testcase-url ${JENKINS_URL}/job/test-override    \
+                    --testrun-url ${blueocean_url}                     \
+                    --outcome ${outcome} --advisory ${bodhi_update_id} \
+                    --stream ${params.STREAM}
+                """)
+            }
+            def bodhi_url="https://bodhi.fedoraproject.org/updates/${bodhi_update_id}"
+            pipeutils.matrixSend("${emoji} ${descPrefix} - [🌊](${blueocean_url}) [🪷](${bodhi_url})")
```
Comment (dustymabe) [inline jobs/test-override.Jenkinsfile:271]: note that descPrefix here will expand to all RPMs in the update. i.e. for https://bodhi.fedoraproject.org/updates/FEDORA-2025-f8be7978e3 it was:

```
clevis-pin-tpm2-0.5.3-9.fc41 dbus-parsec-0.5.0-3.fc41 envision-2.0.0-4.20241209git2.0.0.fc41 fido-device-onboard-0.5.1-2.fc41 gotify-desktop-1.3.7-4.fc41 keylime-agent-rust-0.2.7-4.fc41 keyring-ima-signer-0.1.0-17.fc41 libkrun-1.10.1-2.fc41 python-cryptography-43.0.0-4.fc41 rust-afterburn-5.7.0-3.fc41 rust-cargo-vendor-filterer-0.5.17-2.fc41 rust-coreos-installer-0.23.0-2.fc41 rust-crypto-auditing-agent-0.2.3-2.fc41 rust-eif_build-0.2.1-3.fc41 rust-gst-plugin-reqwest-0.13.3-3.fc41 rust-nu-0.99.1-7.fc41 rust-oo7-cli-0.3.3-4.fc41 rust-openssl-0.10.70-1.fc41 rust-openssl-sys-0.9.105-1.fc41 rust-pore-0.1.17-5.fc41 rust-rpm-sequoia-1.7.0-5.fc41 rust-sequoia-keyring-linter-1.0.1-10.fc41 rust-sequoia-octopus-librnp-1.10.0-6.fc41 rust-sequoia-policy-config-0.7.0-3.fc41 rust-sequoia-sop-0.36.0-3.fc41 rust-sequoia-sq-1.1.0-4.fc41 rust-sequoia-sqv-1.2.1-6.fc41 rust-sevctl-0.6.0-4.fc41 rust-snphost-0.5.0-3.fc41 rust-tealdeer-1.7.1-3.fc41 rustup-1.27.1-6.fc41 s390utils-2.35.0-2.fc41
```

I'm not sure if this is what we want or if we want to filter it down to just the list of RPMs we are triggering on.

cc @cverna

**Comment by dustymabe** [comment]
I think this is ready for some review now.

File: jobs/test-override.Jenkinsfile (hunk 1 above)
Comment (dustymabe) [reply jobs/test-override.Jenkinsfile:299]: I updated this to now just show the RPMs we care about.

File: jobs/test-override.Jenkinsfile [hunk 2]
```diff
@@ -99,6 +106,22 @@ try {
...
 
+    if (params.REPORT_TO_RESULTSDB) {
+        stage("Report Running") {
+            withCredentials([usernamePassword(credentialsId: 'resultsdb-auth',
+                                              usernameVariable: 'RDB_USERNAME',
+                                              passwordVariable: 'RDB_PASSWORD')]) {
+                shwrap("""/usr/lib/coreos-assembler/resultsdb-report    \
+                    --testcase cosa.build-and-test                      \
+                    --testcase-url ${JENKINS_URL}/job/test-override     \
+                    --testrun-url ${JENKINS_URL}/job/test-override      \
```
Comment (jlebon) [inline jobs/test-override.Jenkinsfile:124]: Since we're now doing this from the `test-override` job itself, we can now directly pass the BlueOcean URL even here.

File: jobs/test-override.Jenkinsfile [hunk 3]
```diff
@@ -237,4 +260,36 @@ try {
...
+            // that'll warn and Greenwave/Bodhi treats NEEDS_INSPECTION outcomes
+            // as blocking
+            if (currentBuild.result == 'SUCCESS' || currentBuild.result == 'UNSTABLE') {
+                emoji = "🟢"
+                outcome = 'PASSED'
+            } else {
+                emoji = "🔴"
+                outcome = 'FAILED'
+            }
+            def blueocean_url = "${env.RUN_DISPLAY_URL}"
```
Comment (jlebon) [inline jobs/test-override.Jenkinsfile:286]: Minor/optional: probably not worth a variable?

File: jobs/test-override.Jenkinsfile [hunk 4]
```diff
@@ -35,6 +35,9 @@ properties([
         booleanParam(name: 'ALLOW_KOLA_UPGRADE_FAILURE',
                      defaultValue: false,
                      description: "Don't error out if upgrade tests fail (temporary)"),
+        booleanParam(name: 'REPORT_TO_RESULTSDB',
+                     defaultValue: false,
+                     description: "Report results to resultsdb for the bodhi update"),
```
Comment (jlebon) [inline jobs/test-override.Jenkinsfile:44]: Minor/optional: s/bodhi/Bodhi/

File: jobs/bodhi-trigger.Jenkinsfile [hunk 5]
```diff
@@ -115,7 +115,18 @@ if (matching_streams.size() == 1) {
     error("multiple non-devel matching streams found: ${matching_streams}")
 }
 
-currentBuild.description = msg.update.builds[0].nvr
+// Find the builds in the update we care about and update the
+// description with those build NVRs.
```
Comment (jlebon) [inline jobs/bodhi-trigger.Jenkinsfile:119]: I guess it's a bit more nuanced than that because the Bodhi update might have other packages that are in FCOS but not explicitly listed in `bodhi-testing.yaml`. It'd be confusing if we then fail because of that package and not the visible package in the build description. Agree it's a corner-case though.

File: jobs/test-override.Jenkinsfile [hunk 6]
```diff
@@ -32,6 +32,10 @@ properties([
                description: 'Space-separated testiso test patterns to use (or "skip" to skip)',
                defaultValue: "",
                trim: true),
+        string(name: 'DESCRIPTION',
+               description: 'Description to use when reporting results in Jenkins',
```
Comment (jlebon) [inline jobs/test-override.Jenkinsfile:36]: Minor/optional: in Jenkins and Matrix?

**Review by jlebon — APPROVED** [review]
Thanks! Looks sane to me overall.

File: jobs/test-override.Jenkinsfile (hunk 3 above)
Comment (dustymabe) [reply jobs/test-override.Jenkinsfile:286]: I'll leave the variable in here. It helps my brain understand what `RUN_DISPLAY_URL` is. Otherwise it's not immediately clear to me that it is the blue ocean URL.

File: jobs/bodhi-trigger.Jenkinsfile (hunk 5 above)
Comment (dustymabe) [reply jobs/bodhi-trigger.Jenkinsfile:119]: honestly I think we should be triggering on all FCOS SRPMS.

I think it would have caught something like this https://github.com/coreos/fedora-coreos-tracker/issues/1876, which, yeah, we don't have gating so it still would have got in, but at least we know about it sooner. i.e. just getting a build built without even running tests is useful.

File: jobs/test-override.Jenkinsfile (hunk 6 above)
Comment (dustymabe) [reply jobs/test-override.Jenkinsfile:36]: Updated

File: jobs/test-override.Jenkinsfile (hunk 2 above)
Comment (dustymabe) [reply jobs/test-override.Jenkinsfile:124]: will do

File: jobs/test-override.Jenkinsfile (hunk 4 above)
Comment (dustymabe) [reply jobs/test-override.Jenkinsfile:44]: will fix
